src/python_pachyderm/mixin/pfs.py

```diff
@@ -111,11 +71,28 @@ def read(self, size: int = -1) -> bytes:
         bytes
             Content from the stream.
         """
-        return self._file.read(size)
+        try:
+            if size == -1:
+                # Consume the entire stream.
+                for message in self._stream:
```

Comment:
Assuming an infinite stream (or absurdly large stream), should we cap how much we read in before returning so this isn't stuck in an infinite loop?

Reply:
That's a good question, I would say that it's fine and I'd rather just be consistent with how `.read()` works with standard python file objects.

Also fun fact, I tried this to see if python would let me do it and I just hard crashed my computer running out of memory:
```python
with open("/dev/urandom") as evil:
    evil.read()
```